src/app/hooks/useGlean.ts

```diff
@@ -4,46 +4,116 @@
 
 "use client";
 
-import { useEffect } from "react";
+import { useEffect, useState } from "react";
 import Glean from "@mozilla/glean/web";
-import * as pageEvents from "../../telemetry/generated/page";
+// @ts-expect-error We do not have type declarations for metrics.yaml, yet.
```

Comment:
If you open `next-env.d.ts`, then command-click `next/image-types/global`, you'll see an example of how you can type this in a `.d.ts` file:

```ts
declare module '*.yarm' {
  type GleanExtraKeys = Record<string, string | boolean | number>;

  type GleanCategoryKey = string | number | symbol;

  type GleanEvent = {
    [key: string]: {
      type: string;
      description: string;
      bugs: string;
      data_reviews: string;
      data_sensitivity: string;
      notification_emails: string;
      expires: string;
      extra_keys: GleanExtraKeys;
      record: (params: GleanExtraKeys) => void;
    };
  };

  type GleanCategories = {
    [key: GleanCategoryKey]: GleanEvent;
  };

  type GleanMetrics = {
    $schema: string;
  } & GleanCategories;

  const content: GleanMetrics;

  export default content;
}
```

Reply:
Glean.js supports typescript but our `package.json` has it outputting JS:
https://github.com/mozilla/blurts-server/blob/200db634df253ec020fdc892036826a5d694ed12/package.json#L27

If you change the `-o javascript` to `-o typescript` then it should produce `.ts` files instead which should have type definitions.

Reply:
Good to know that we have TS available as an output format. I’ve changed the `build-glean` script accordingly in [3f81092](https://github.com/mozilla/blurts-server/pull/3813/commits/3f81092e160e5765e41c267ac390fdb41e13a02a).